core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/util/DatabaseUtil.java

```diff
@@ -347,28 +357,25 @@ public static int getIntegerValueFromDatabase(Connection dbConnection, String sq
                     }
                 }
             }
-            rs = prepStmt.executeQuery();
-            if (rs.next()) {
-                value = rs.getInt(1);
+            try(ResultSet rs = prepStmt.executeQuery()) {
+                if (rs.next()) {
+                    value = rs.getInt(1);
+                }
+                return value;
             }
-            return value;
         } catch (SQLException e) {
             String errorMessage = "Using sql : " + sqlStmt + " " + e.getMessage();
             if (log.isDebugEnabled()) {
                 log.debug(errorMessage, e);
             }
             throw new UserStoreException(errorMessage, e);
-        } finally {
-            DatabaseUtil.closeAllConnections(null, rs, prepStmt);
```

Comment:
The try-with-resource pattern automatically handles the ResultSet and PreparedStatement objects, but the Connection, "dbConnection" in this scope, still needs to be closed.